lib/ecto/migration.ex

```diff
@@ -210,6 +210,13 @@ defmodule Ecto.Migration do
 
           config :app, App.Repo, migration_repo: App.MigrationRepo
 
+    * `:migration_cast_version_column` - Ecto uses a `version` column of type
+      `bigint` for the underlying migrations table (usually `schema_migrations`). By
+      default, Ecto casts this column as an `int` when reading or writing to the database
+      when running migrations. This behavior is there for legacy reasons, but if you
+      want to disable the `CAST`ing, you can set this option to `false`. You might want
+      to do that if you are using a database that doesn't support `CAST`ing in all queries.
```

Comment:
I think we can default to false and those on old databases can enable it again?

Reply:
Oh, I remember now. Rails uses strings, we always used integer, and this was added for compatibility. Yeah, we can make this behaviour opt-in.

Reply:
@josevalim awesome, nice! I've pushed changes to do that.